scipy/integrate/_ode.py

```diff
@@ -384,8 +384,8 @@ def set_integrator(self, name, **integrator_params):
         if integrator is None:
             # FIXME: this really should be raise an exception. Will that break
             # any code?
-            warnings.warn('No integrator name match with %r or is not '
-                          'available.' % name)
+            message = f'No integrator name match with {name!r} or is not available.'
+            warnings.warn(message, stacklevel=3)
```

Comment:
I'm not sure that it matters, but this [is a public method](https://docs.scipy.org/doc/scipy/reference/generated/scipy.integrate.ode.set_integrator.html#scipy.integrate.ode.set_integrator).
```suggestion
            warnings.warn(message, stacklevel=2)
```
It looks like sometimes it's used via other functions, but still I guess it should be `2` since it's public..